src/main/java/at/hannibal2/skyhanni/features/rift/area/mountaintop/SunGeckoHelper.kt

```diff
@@ -0,0 +1,284 @@
+            if (onFirstPhase) displayHealthLeft += totalHealth
+            displayTotalHealth *= 2
+        }
+
+        var healthColor = "§a"
+        if (displayHealthLeft <= displayTotalHealth / 2) {
+            healthColor = "§e"
+        }
+
+        val health = "$healthColor$displayHealthLeft§f/§a$displayTotalHealth§c❤"
+        display.add("§eSun Gecko $health")
+        display.add("$actionBarFormatted §e§lCombo: x$combo")
+
+        // this is just a total guess but it looks right enough
+        // i think its inconsistent because of how often the action bar updates
+        var expiryTime = 5.seconds + 200.milliseconds
+        if (modifiers.contains(Modifiers.COLLECTIVE)) {
+            expiryTime += (modifiers.size * 200).milliseconds
+        }
+        val timeLeft = timeSinceLastHit + expiryTime
+        if (timeLeft.timeUntil().inWholeMilliseconds > expiryTime.inWholeMilliseconds - 800.milliseconds.inWholeMilliseconds ||
+            timeLeft.isInPast()) {
+            display.add("§aCombo Timer: ${expiryTime.format()}/${expiryTime.format()}")
+        } else {
+            display.add("§aCombo Timer: ${timeLeft.timeUntil().format(showMilliSeconds = true)}/${expiryTime.format()}")
+        }
+
+        if (config.showModifiers) {
+            display.add("§6Modifiers:")
+            if (modifiers.isEmpty()) {
+                display.add("§eNone")
+            }
+            for (modifier in modifiers) {
+                display.add("§e${modifier.formattedName}")
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onMobSpawn(event: MobEvent.Spawn) {
+        if (!event.mob.name.contains("Sun Gecko")) return
+        if (event.mob.name.contains("?") && config.highlightFakeBoss) {
+            event.mob.highlight(Color.RED.addAlpha(80))
+            /* RenderLivingEntityHelper.setEntityColorWithNoHurtTime(
+                event.mob.baseEntity,
+                Color.RED.addAlpha(80),
+            ) { config.highlightFakeBoss } */
+        } else {
+            if (config.highlightRealBoss) {
+                event.mob.highlight(Color.GREEN.addAlpha(80))
+                /* RenderLivingEntityHelper.setEntityColorWithNoHurtTime(
+                    event.mob.baseEntity,
+                    Color.GREEN.addAlpha(80),
+                ) { config.highlightRealBoss } */
+            }
+            if (currentBoss == null) {
+                currentBoss = event.mob
+            } else {
+                if (currentBoss?.baseEntity?.isEntityAlive == false ||
+                    (currentBoss?.health?.toInt() ?: 0) < 20) {
+                    currentBoss = event.mob
+                }
+            }
+        }
+
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!isEnabled() || !inTimeChamber) return
+
+        updateDisplay()
+
+        if (currentBoss?.baseEntity?.isEntityAlive == false) {
+            currentBoss = null
+        }
+
+        if (currentBoss == null) return
```

Comment:
```suggestion
val currentBoss = currentBoss ?: return
```

that way you dont have to deal with currentBoss being nullable underneath